src/components/Swap/SwapSettings.tsx

```diff
@@ -75,36 +83,47 @@ const SwapSettings = ({
             <Field
               name="minDexFee"
               validate={(value) => {
-                return validateNumber(value, {
-                  maxDecimals: ERG_DECIMALS,
-                });
+                return validateMinDexFee(value);
               }}
             >
               {(props: FieldRenderProps<string>) => (
-                <Input
-                  placeholder="0.0"
-                  width="100%"
-                  label={ERG_TOKEN_NAME}
-                  {...props.input}
-                  value={
-                    Number(minDexFee) === 0
-                      ? minDexFee
-                      : renderFractions(BigInt(minDexFee), ERG_DECIMALS)
-                  }
-                  onChange={({ currentTarget }) => {
-                    const value = toFloat(currentTarget.value, ERG_DECIMALS);
+                <>
+                  <Row>
+                    <Input
+                      placeholder="0.0"
+                      width="100%"
+                      label={ERG_TOKEN_NAME}
+                      {...props.input}
+                      value={
+                        Number(minDexFee) === 0
+                          ? minDexFee
+                          : renderFractions(BigInt(minDexFee), ERG_DECIMALS)
+                      }
+                      onChange={({ currentTarget }) => {
```

Comment:
This callback is better to place outside of render returning value to increase readability.